src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +10,33 @@ public class ReadWriteList<E> {
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        lock.writeLock().lock();
+        try {
+            list.add(element);
+        } finally {
+            lock.writeLock().unlock();
+        }
```

Comment:
would be nice to have read/write locks also as fields to not get them each time

Reply:
added